docs/DeveloperGuide.md

```diff
@@ -2,37 +2,159 @@
+user tries to add company information.
+
+### [DONE] Add feature
+#### Implementation
+The proposed <code>CompanyList.add</code> is facilitated by <code>AddCommand</code>. It will add new company to existing 
+list of companies. It has three parameters, <code>String companyName </code>, <code>Int contactNumber </code>, and <code>
+String contactEmail</code>. Using this information, <code>CompanyList.add</code> creates new Company object and adds it 
+at the end of the company list. After successfully executing the command, it will show successful addition message using
+the related method in the <code>Ui</code>.
+
+### [DONE] Delete feature
+#### Implementation
+The proposed <code>CompanyList.deleteCompanyInformation(index)</code> is facilitated by <code>DeleteCommand</code>. 
+It will delete the company at the specified index in the list of companies. After successfully executing the command, 
+it will show successful deletion message using the related method in the <code>Ui</code>.
+
+### [DONE] Storing event details feature
+#### Implementation
+The proposed eventDetailStorage feature will allow users to save the event details of the current session to hard disk. 
+A text file will be created to store information such as event name and venue information. This text file will be 
+overwritten everytime the user makes changes to the stored information, such as renaming the event name or changing the 
+venue of the event. When relaunching EventUS, the program will check if there is a text file which contains the 
+information of an event. If the text file exists, the information of the Event instance will be updated with the event 
+details. If there is no text file, a text file will be created.
+
+### [DONE] List companies feature
+#### Implementation
+The proposed <code>CompanyList.printCompanyInformation</code> is facilitated by <code>ListCompanyCommand</code>. It will
+print out the companies that are stored in an ArrayList called companyList. The Company object is then printed out with 
+an integer denoting its index, four parameters, <code>String companyName </code>, <code>String industry </code>, 
+<code>Int contactNumber </code>, and <code>String contactEmail </code>, and its confirmation status. It will print out 
+each company from the entire companyList from index 0 to the final index of the last company stored in companyList.
+
+### [DONE] Store Company information to text file feature
+#### Implementation
+The proposed <code>CompanyListEncoder.write()</code> feature checks if a text file of name "companyList.txt" exists.
+If the text file does not exist, it is created when the programme boots up.
+If the text file exists, this method iterates through the list of company information in the list of companies and 
+stores the companies' information in the text file. Prior to writing the company information to this text file, all 
+previously saved text in this file is overwritten.
+![Encoder.png](UML%2FImage%2FEncoder.png)
```

Comment:
Most crucial mistake is, the `alt` block is supposed to have an else block to tell the reader what will happen if the condition is not satisfied. If nothing happens in the else block, maybe you might wanna consider using an `opt` block instead

Reply:
Another thing is, most of the Return line in the sequence diagram does not have any name given to it. For instance, the `getCompanyName()` and `getContactNumber()` by right should return the `companyName` and `contactNumber` but it is not reflected in the sequence diagram.

Reply:
In the description it is mentioned that the `txt` file will be created if it does not exist, and the content will be overwritten if it already exists. But this is not reflected in the sequence diagram at all.

Reply:
Also, the lifetime of the activation bar of the `Company` does not seem accurate. Basically, `Company` object does not only exist when you call `getCompanyName()`